src/app/inventory/store/sockets.ts

```diff
@@ -479,10 +484,51 @@ function buildSocket(
     socketIndex: index,
     plugged,
     plugOptions,
+    plugSet: socketDef.reusablePlugSetHash
+      ? buildCachedDimPlugSet(defs, socketDef.reusablePlugSetHash)
+      : undefined,
     curatedRoll,
     hasRandomizedPlugItems,
     reusablePlugItems: reusablePlugs,
     isPerk,
     socketDefinition: socketDef,
   };
 }
+
+// This cache is used to reuse DimPlugSets across items. If we didn't do this each
+// item would have their own instances of shaders, which is 100's of plugs.
+let reusablePlugSetCache: { [plugSetHash: number]: DimPlugSet | undefined } = {};
+
+/**
+ * Resets the cache we use to ensure plug sets are only created once per hash.
+ */
+export function resetReusablePlugSetCache() {
+  reusablePlugSetCache = {};
+}
+
+/**
+ * This builds a DimPlugSet based off hash passed in, that links to a DestinyPlugSetDefinition.
+ * We cache values so if a plug set is present on more than one item, all those items will share
+ * the plug set instance.
+ */
+function buildCachedDimPlugSet(defs: D2ManifestDefinitions, plugSetHash: number) {
```

Comment:
```suggestion
function buildCachedDimPlugSet(defs: D2ManifestDefinitions, plugSetHash: number): DimPlugSet {
```
as a foundational item building function let's add an explicit return type here